### ISSUE TITLE
[Feature request] Drag and drop creates link file. gsettings entry to disable this behaviour.

### ISSUE THREAD

**Issue by Blast-City** [elementary/files#921]
1- Drag drop a file or folder a couple of pixels away.
**Result:** A link to the file or folder is created.

I think there should be a gsettings entry to disable this feature. Sometimes you move a file or folder by accident and a link is created.

P.S: I'm sorry if this has been discussed before.

**Comment by jeremypw** [comment]
This is readily undone by pressing `<Ctrl>Z` so I am not sure the UX team would approve the creation of a new setting. It may be better to make the default behaviour to show a context menu (like dragging with the secondary button does).

**Comment by Blast-City** [comment]
> This is readily undone by pressing `<Ctrl>Z` so I am not sure the UX team would approve the creation of a new setting. It may be better to make the default behaviour to show a context menu (like dragging with the secondary button does).

Hi.

I don't see why it should be enabled If by dragging with the right click does the same thing and even show the context menu (sort of a confirmation).

When you drag a file or folder in the same location it should show the animation that appears when you press esc while holding the file and do nothing.

While testing this issue I also found out you can't drag with the right click when a file/folder is previously selected. Is this a known issue?

Thanks.

**Comment by jeremypw** [comment]
Thanks for the suggestions - we'll see what the UX team say.  I disagree that the "invalid drop" animation should be shown as it not an invalid drop location.

There are some bugs with DnD and there is an outstanding PR that fixes some of them.  I find that drag with secondary on a selected item sometimes (not always) fails but a second attempt usually succeeds.

**Comment by Blast-City** [comment]
> Thanks for the suggestions - we'll see what the UX team say. I disagree that the "invalid drop" animation should be shown as it not an invalid drop location.
> 
> There are some known bugs with DnD and there is an outstanding PR (#814) that fixes some of them. I find that drag with secondary on a selected item sometimes (not always) fails but a second attempt usually succeeds.

Oh. Ok. That makes sense.

Thanks.

**Comment by jeremypw** [comment]
Note that with the secondary button you have to start to drag within a pretty short time else it assumes you are going to show the context menu and blocks DnD.

**Comment by donadigo** [comment]
Besides adding additional settings I also think that the behaviour is confusing. Many times I will start dragging a file and will be like "actually no" and will drag back the file to it's original position only to create a link that I didn't want to create. I think that creating a link should happen with dragging the file only with the secondary button and when dragging with primary, nothing should happen.

**Comment by jeremypw** [comment]
I have had the same issue but I just drop the file outside the window e.g. on wingpanel to cancel. I agree this is not ideal.

**Comment by goodman65** [comment]
I think most of the time, users don't like using mouse and keyboard at the same time. So it's just natural that users expect drag-and-drop to be canceled in the same location or with a right-click. And that's what other OS or DE do. I really like elementary, but this is one of the points that depresses me. I wish to see this changed soon in elementary 6.

**Comment by jeremypw** [comment]
I agree that dropping a file onto itself should do nothing. Nautilus appears to do nothing if dropped into the same directory (effectively we are moving it into the same place so this seems logical). So I am happy to remove the create link function from the primary DnD if the design team agrees.  My only concern is that secondary drag is not very discoverable as, unless done quickly, it is preempted by the context menu.

**Comment by jeremypw** [comment]
I note Dolphin shows a menu giving options to Copy or Link which seems sensible.  There is no reason to prefer one action over the other and perform it automatically.  The alternative is to use modifier keys but this is inapplicable to touchscreens and not easily discoverable.
